ConnectionFactory.php

```diff
@@ -43,6 +44,22 @@ public function createConnection(array $params, Configuration $config = null, Ev
 
         $connection = DriverManager::getConnection($params, $config, $eventManager);
 
+        if (! isset($params['pdo']) && ! isset($params['charset'])) {
+            $params            = $connection->getParams();
+            $params['charset'] = 'utf8';
+            $driver            = $connection->getDriver();
+
+            if ($driver instanceof AbstractMySQLDriver) {
+                $params['charset'] = 'utf8mb4';
+
+                if (! isset($params['defaultTableOptions']['collate'])) {
+                    $params['defaultTableOptions']['collate'] = 'utf8mb4_unicode_ci';
+                }
+            }
+
+            $connection = new $connection($params, $driver, $connection->getConfiguration(), $connection->getEventManager());
```

Comment:
I'd rather use `DriverManager::getConnection` again rather than assuming the constructor argument here

Reply:
That's the "contract" of all connection classes: the wrapper_class parameter already requires this signature.